Remove the old morphic-based FileList

hernanmd opened the pull request:
This is just a first PR to remove the FileList based on Morphic. The tool has already a replacement version based in Spec 2 recently integrated and ready to be used.
The new file system lister or browser is implemented in `StFileSystemPresenter`.

Commit:
First commit to remove the old morphic-based FileList

Commit:
Updating UITheme to use new file dialogs

guillep reviewed (changes requested):
Hi @hernanmd Thanks!

This PR is just removing the tool from the listed packages, but it's not removing the package. To do that in Iceberg you need to use the "Remove from repository" option and commit that.

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into remove_old_filelist

Commit:
Two more UITheme methods to check against new File Dialogs

Commit:
Merge a249462f896902102a0c5249e99685d64b81be40

hernanmd commented:
> Hi @hernanmd Thanks!
> 
> This PR is just removing the tool from the listed packages, but it's not removing the package. To do that in Iceberg you need to use the "Remove from repository" option and commit that.

Thanks, I will fix this today as there are a couple of users of UITheme that needs translation.

guillep commented:
> > Hi @hernanmd Thanks!
> > This PR is just removing the tool from the listed packages, but it's not removing the package. To do that in Iceberg you need to use the "Remove from repository" option and commit that.
> 
> Thanks, I will fix this today as there are a couple of users of UITheme that needs translation.

Oh I see this is a work in progress. I'll mark the PR as draft to avoid accidental merges :)

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into remove_old_filelist

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into remove_old_filelist

Commit:
Adapt remaining use cases of UITheme for file dialogs.
This commit uses StFileBrowserTextBasedContentPreviewer new feature which enables to adaptively change previewers.

Commit:
Merge bb9e2e5690890e6e20c2e4293284fef868ae0e6f

hernanmd commented:
Updated. This needs https://github.com/pharo-spec/NewTools/pull/682 prior to be applied.

jecisc committed:
Merge branch 'Pharo12' into remove_old_filelist

jecisc commented:
I resolved the conflict. The PR in NewTools got merged so maybe this PR should not be a draft anymore?

demarey commented on src/Polymorph-Widgets/UITheme.class.st:
```diff
@@ -754,12 +754,12 @@ UITheme >> chooseColorIn: aThemedMorph title: aString color: aColor for: aBlock
-	^ dialog openModal answer
+	^ StOpenDirectoryDialog new
+		defaultFolder: path;
+		title: (title ifNil: [ 'Choose Directory' translated ]);
```
This does not work.
Title if provided is not displayed as the window is not yet open

hernanmd replied:
It looks buggy certainly. I don't know because I never use `translate`  :)
Anyway this could be soon deprecated, @jecisc ?

jecisc replied:
translated should return a string translated in the local language if the translation project is loaded IIRC. In the worst case it should return the original string in any case.

jecisc replied:
I think that some of the API should get deprecated but I'm not yet sure what will get deprecated exactly and right now I'm working on Moose

demarey replied:
The problem here is not the translation. Setting the title before the window is open has no effect.
You can try by yourself:
```smalltalk
StOpenDirectoryPresenter new
		title: 'test set tittle';
		openModal
```

hernanmd replied:
I've sent a PR in https://github.com/pharo-spec/NewTools/pull/716 to fix this

jecisc replied:
Merged

Ducasse commented:
As discussed for P13

jecisc commented:
This has been done in other PRs
